### PR TITLE
Add batch API support

### LINKED ISSUE #115: [Request] Batch API: Return multiple assets from a query
Is it possible for a query to return multiple assets? Can the asset variable be a multiselect?

### PR DESCRIPTION
**Pull request by toddtreece**
Adds support for batching multiple asset requests.

Fixes #115 

image

### CODE REVIEW

File: src/sitewiseCache.ts [hunk 1]
```diff
@@ -174,13 +174,13 @@ export class SitewiseCache {
...
-        description: asset.arn,
-      });
-    }
+    //for (const asset of this.assetsById.values()) {
+    //  options.push({
+    //    label: asset.name,
+    //    value: asset.id,
+    //    description: asset.arn,
+    //  });
+    //}
```
Comment (toddtreece) [inline src/sitewiseCache.ts:183]: i'm not clear why this is needed

File: pkg/testdata/property-value-empty-A.golden.jsonc [hunk 2]
```diff
@@ -1,57 +1,6 @@
...
-      "data": {
-        "values": [
-          [],
-          [],
-          []
-        ]
-      }
-    }
-  ]
+  "frames": []
```
Comment (ryantxu) [inline pkg/testdata/property-value-empty-A.golden.jsonc:5]: is this change expected?

File: pkg/testdata/property-history-values.json [hunk 3]
```diff
@@ -1,459 +1,464 @@
...
-                "OffsetInNanos": 0,
-                "TimeInSeconds": 1612207200
-            },
-            "Value": {
-                "BooleanValue": null,
-                "DoubleValue": 23.81007059955162,
-                "IntegerValue": null,
-                "StringValue": null
-            }
+    "SuccessEntries": [
```
Comment (ryantxu) [inline pkg/testdata/property-history-values.json:2]: Looks like the response format change significantly... is this expected?

Comment (toddtreece) [reply pkg/testdata/property-history-values.json:2]: yep, this is the structure of the batch API response from AWS

File: pkg/testdata/property-value-empty-A.golden.jsonc (hunk 2 above)
Comment (toddtreece) [reply pkg/testdata/property-value-empty-A.golden.jsonc]: updated

**Review by ryantxu — APPROVED** [review]

File: pkg/framer/property_aggregate.go [hunk 4]
```diff
@@ -90,23 +90,56 @@ func addAggregateFieldValues(idx int, fields map[string]*data.Field, aggs *iotsi
...
+		}
+		frame.Meta = &data.FrameMeta{
+			Custom: models.SitewiseCustomMeta{
+				NextToken:  aws.StringValue(resp.NextToken),
+				Resolution: aws.StringValue(a.Request.Entries[i].Resolution),
+				Aggregates: aws.StringValueSlice(a.Request.Entries[i].AggregateTypes),
+			},
+		}
+
+		for _, e := range resp.ErrorEntries {
```
Comment (idastambuk) [inline pkg/framer/property_aggregate.go:114]: Should this block be outside the success entries block?

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,10 @@
 
 All notable changes to this project will be documented in this file.
 
+## v1.6.0
```
Comment (idastambuk) [inline CHANGELOG.md:5]: Should be updated, we've had two patch versions since and an addition to main since the last patch :)

**Pull request merged**